src/api/app/jobs/send_event_emails.rb

```diff
@@ -33,4 +40,13 @@ def create_rss_notifications(event)
       )
     end
   end
+
+  def needs_logging?(event)
```

Comment:
That could work too but IMO its simpler to check the if the class is descendant, especially since this is how the logic has been already for a while so its unlikely to change significantly so I think using inheritance for this method would be overkill..